docs/guides/service_configuration_file.md

```diff
@@ -0,0 +1,304 @@
+   ```
+   The corresponding environment variable `SERVICE_HELLO_WORLD_HELLO_WORLD_STRING`will be read by the agent in runtime. If not found, the default value `HI_WORLD!` will be taken by the agent.
+
+!!! note
+    Observe that the environment variable associated to an overridden attribute has the form  `<COMPONENT_TYPE>_<UPPERCASE_ATTRIBUTE_PATH>`, since they are generated from the `json` path to the attribute. Moreover, we recommend using the overridden format
+    ```yaml
+    <overridden_attribute>: ${<ENVIRONMENT_VARIABLE>:<type>:<default_value>}
+    ```
+    as in the example above to fallback into a safe default value in case the environment variable is not defined.
+
+### Multiple overrides
+
+In the service configuration file, you can override different values for different agents using the multiple override feature, using the pattern below:
+
+```yaml
+(...)
+---
+public_id: valory/hello_world_abci:0.1.0
+type: skill
+<agent_number>:
+  <overridden_attributes>
+```
+
+For example, if we wish that each agent in the [Hello World service](../demos/hello_world_demo.md) outputs a different message, we can achieve this as follows:
+
+```yaml
+(...)
+---
+public_id: valory/hello_world_abci:0.1.0
+type: skill
+0:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 0
+1:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 1
+2:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 2
+3:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 3
+```
+
+!!! warning "Important"
+    If you choose to use the multiple override feature, you have to define overrides for **every** single agent that composes the service.
+
+If you have repetitive overridden parameters, you can define them using [YAML anchors](https://yaml.org/spec/1.2.2/) under the section `extra`, and reference them when defining overrides for different agents.
+
+```yaml
+(...)
+---
+extra:
+  benchmark_tool:
+    args: &id001
+      log_dir: /benchmarks
+public_id: valory/hello_world_abci:0.1.0
+type: skill
+0:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 0
+    benchmark_tool:
+      args: *id001
+1:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 1
+    benchmark_tool:
+      args: *id001
+2:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 2
+    benchmark_tool:
+      args: *id001
+3:
+  models:
+    params:
+      args:
+        hello_world_message: Hello, from agent 3
+    benchmark_tool:
+      args: *id001
+```
+
+
+### Overrides with complicated data structures
+
+If you have nested list the environment export rules will differ as per the inner data structures. If the list is a strict list, meaning the inner values are either primitive data types like `int`, `str`, `float` or `boolean`, the list will be exported as a `json` string.
```

Comment:
First time export is mentioned in the documentation. It should be mentioned what the "export" means before explaining the difference between exporting one data type or another.

Reply:
`export ` is a pretty common word, I don't think we need to define what environment exports are

Reply:
I think it is clear what export to environment means yes, the fact that this is happening in itself is not obvious though and is mentioned nowhere else in the documentation.

Reply:
@jmoreira-valory I suggest you open a ticket to add section on how env var export and parsing works, we can have a quick call and I'll explain the process, then you can add the section